### PR TITLE
fix: OOM when eth_getLogs response too large

### PR DESCRIPTION
**Pull request by yihuang**
## Description

backport: https://github.com/tharsis/ethermint/pull/860

- add limit to number of logs of filter response
- make block limit and log limit configurable

______

For contributor use:

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/cosmos-sdk/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added  relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer

______

For admin use:

- [ ] Added appropriate labels to PR (ex. `WIP`, `R4R`, `docs`, etc)
- [ ] Reviewers assigned
- [ ] Squashed all commits, uses message "Merge pull request #XYZ: [title]" ([coding standards](https://github.com/tendermint/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

File: rpc/ethereum/namespaces/eth/filters/filters.go [hunk 1]
```diff
@@ -145,36 +144,37 @@ func (f *Filter) Logs(_ context.Context) ([]*ethtypes.Log, error) {
 	from := f.criteria.FromBlock.Int64()
 	to := f.criteria.ToBlock.Int64()
 
-	blocks, err := f.backend.GetFilteredBlocks(from, to, f.bloomFilters, len(f.criteria.Addresses) > 0)
-	if err != nil {
-		return nil, err
-	}
+	for height := from; height <= to; height++ {
+		bloom, err := f.backend.BlockBloom(&height)
```
Comment (yihuang) [inline rpc/ethereum/namespaces/eth/filters/filters.go:148]:
```suggestion
	for i := from; i <= to; i++ {
	        height := i
		bloom, err := f.backend.BlockBloom(&height)
```

**Review by tomtau — DISMISSED** [review]
there's some formatting error, but it looks ok to be ported

**Comment by yihuang** [comment]
> there's some formatting error, but it looks ok to be ported

formatting issue fixed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -35,7 +35,11 @@ Ref: https://keepachangelog.com/en/1.0.0/
 
 # Changelog
 
-# Unreleased
+## Unreleased
+
+### Bug Fixes
+
+* (rpc) [tharsis#860](https://github.com/tharsis/ethermint/pull/860) Fix `eth_getLogs` when specify blockHash without address/topics, and limit the response size.
```
Comment (yihuang) [inline CHANGELOG.md:42]:
```suggestion
* (rpc) [crypto-org-chain#42](https://github.com/crypto-org-chain/ethermint/pull/42) Fix `eth_getLogs` when specify blockHash without address/topics, and limit the response size.
```

**Comment by yihuang** [comment]
I've pushed directly, closing now.

**Pull request closed without merging**